src/ServiceInsight/ServiceControl/DefaultServiceControl.cs

```diff
@@ -533,4 +532,14 @@ public PresentationHint Hint
             set { hint = value; }
         }
     }
+
+    public class ResponseUnicodeCleaner
```

Comment:
What's the problem with a 'whole new class'? Cleaning unicode characters is not a responsibility of the ServiceControl class. Moving it out in favor of SRP and cleaner code. Happy to extract the class and move it to a different namespace though, if it helps.